### PR TITLE
Clean up EKSA resources

### PR DESCRIPTION
**Pull request by mdsgabriel**
*Issue #, if available:*

*Description of changes:*
Delete EKSA resources when deleting a workload cluster (GitOpsConfig, OIDCCOnfig, VSphereMachineConfig and VsphereDatacenterconfig)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/559#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/clustermanager/cluster_manager.go
Comment (pokearu) [inline pkg/clustermanager/cluster_manager.go:274]: Do we want to add the same support for AWSIamConfig as well? Since it’s also created as a CR on the cluster.

Comment (mdsgabriel) [reply pkg/clustermanager/cluster_manager.go:275]: added

**Comment by abhay-krishna** [comment]
@mdsgabriel: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | 1023e666ebd5744248737636a5cde68e6ad5382d | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/559/eks-anywhere-e2e-presubmit/1455733233541976064) | `/test eks-anywhere-e2e-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=559). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mdsgabriel).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by abhay-krishna** [comment]
@mdsgabriel: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-e2e-presubmit | e5e8a96606e1da1b98f926fcee60db6463b73c7c | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/559/eks-anywhere-e2e-presubmit/1455761583979368448) | `/test eks-anywhere-e2e-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=559). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:mdsgabriel).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: pkg/clustermanager/cluster_manager.go
Comment (mitalipaygude) [inline pkg/clustermanager/cluster_manager.go:1009]: this should probably be delete AWSIAM config

Comment (mdsgabriel) [reply pkg/clustermanager/cluster_manager.go]: done

File: pkg/clustermanager/cluster_manager.go
Comment (jiayiwang7) [inline pkg/clustermanager/cluster_manager.go:262]: i think we need to retry each kubectl call instead of all together

File: pkg/executables/kubectl.go
Comment (mitalipaygude) [inline pkg/executables/kubectl.go:193]: nit: spelling vspheredatacenterconfig

File: pkg/executables/kubectl.go
Comment (mitalipaygude) [inline pkg/executables/kubectl.go:202]: nit: spelling vspheremachineconfig

**Review by mitalipaygude — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/559#" title="Author self-approved">mdsgabriel</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/559#pullrequestreview-797054945" title="Approved">mitalipaygude</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mdsgabriel,mitalipaygude]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/executables/kubectl.go
Comment (mdsgabriel) [reply pkg/executables/kubectl.go]: done

File: pkg/clustermanager/cluster_manager.go
Comment (mdsgabriel) [reply pkg/clustermanager/cluster_manager.go:262]: Added  "--ignore-not-found=true" to delete command making them idempotent.

File: pkg/executables/kubectl.go
Comment (mdsgabriel) [reply pkg/executables/kubectl.go]: fixed

**Pull request merged**
